sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/implementation/Utility.java

```diff
@@ -94,8 +112,120 @@ public HttpHeaders getHeaders() {
     public static Throwable mapToHttpResponseExceptionIfExist(Throwable throwable) {
         if (throwable instanceof TextAnalyticsErrorException) {
             TextAnalyticsErrorException errorException = (TextAnalyticsErrorException) throwable;
-            return new HttpResponseException(errorException.getMessage(), errorException.getResponse());
+            return new HttpResponseException(errorException.getMessage(), errorException.getResponse(),
+                toTextAnalyticsError(errorException.getValue()));
```

Comment:
@srnagar Is this the same thing that management SDK's are doing too?
And you had some opinions to have a new model type for the exception rather than doing it this way?

Reply:
as discussion in the TA meeting, this change looks good.

Reply:
In `toTextAnalyticsError` .NET is using the top-level error to form the exception, but here we are using the innerError value if present.

Reply:
This is what python used to use innerError 

https://github.com/Azure/azure-sdk-for-python/pull/11155/files#diff-6814e3aaeba6362738bfde592d65bd9bR64